### PR TITLE
Disable cache for archive && content endpoint.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1095/
Test PASSed.

**Comment by urosj** [comment]
Needs to be updated and invalidated only in case the charm is public.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1096/
Test PASSed.

File: internal/v4/archive.go [hunk 1]
```diff
@@ -424,6 +424,19 @@ func (h *ReqHandler) serveArchiveFile(id *router.ResolvedURL, fullySpecified boo
 	return errgo.WithCausef(nil, params.ErrNotFound, "file %q not found in the archive", filePath)
 }
 
+func (h *ReqHandler) isPublic(id charm.Reference) bool {
```
Comment (mhilton) [inline internal/v4/archive.go]: This could use the public flag in the base entity

File: internal/v4/archive.go [hunk 2]
```diff
@@ -516,26 +529,22 @@ func verificationError(err error) error {
...
-// The idFullySpecified header specifies whether
-// the entity id in the request was fully specified by the client.
-func setArchiveCacheControl(h http.Header, idFullySpecified bool) {
-	age := archiveCacheVersionedMaxAge
-	if !idFullySpecified {
-		age = archiveCacheNonVersionedMaxAge
-	}
-	seconds := int(age / time.Second)
-	h.Set("Cache-Control", "public, max-age="+strconv.Itoa(seconds))
+// The isPublic parameter pecifies whether
```
Comment (mhilton) [inline internal/v4/archive.go]: *specifies

File: internal/v4/archive_test.go [hunk 3]
```diff
@@ -1532,12 +1531,13 @@ func patchArchiveCacheAges(s interface {
...
 // for the current test.
-func assertCacheControl(c *gc.C, h http.Header, idFullySpecified bool) {
-	seconds := 5
-	if idFullySpecified {
-		seconds = 20
+func assertCacheControl(c *gc.C, h http.Header, isPublic bool) {
+	if isPublic {
+		seconds := 5
+		c.Assert(h.Get("Cache-Control"), gc.Equals, fmt.Sprintf("public, max-age=%d", seconds))
+	} else {
```
Comment (mhilton) [inline internal/v4/archive_test.go:1538]: normally we would save the indent by returning in the clause above and not having the else.

**Comment by mhilton** [comment]
LGTM with a couple of trivials

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1111/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1112/
Test PASSed.

File: internal/v4/archive.go (hunk 2 above)
Comment (fabricematrat) [inline internal/v4/archive.go]: Done.

**Comment by urosj** [comment]
:+1: with QA

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**